### PR TITLE
[pkg/ottl] Minor updates to OTTL readmes

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

I updated a few small things in the OTTL readme files:
- Renamed references to factory functions to Converters.
- Capitalized "Converter" for consistency.
- Moved all non-Converter functions from the "Converters" section to the "Functions" section.
- Minor wording changes

### CODE REVIEW

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -14,14 +14,14 @@ See the component-specific guides for how each uses error mode:
 
 ## Functions
 
-Functions are the way that components that use OTTL transform telemetry.
+Functions are the way that components that use OTTL to transform telemetry.
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:17]:
```suggestion
Functions are what OTTL uses to transform telemetry.
```
that extra `to` didn't sound right in the original sentence.

File: pkg/ottl/ottlfuncs/README.md [hunk 2]
```diff
@@ -444,11 +246,209 @@ Examples:
 
 - `truncate_all(resource.attributes, 50)`
 
+## Converters
+
+Converters are functions that help translate between the OTTL grammar and the underlying pdata structure.
+They manipulate the OTTL grammar value into a form that will make working with the telemetry easier or more efficient.
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:252]: Can you add a sentence in here that Converters never change the underlying telemetry.

File: pkg/ottl/ottlfuncs/README.md (hunk 1 above)
Comment (evan-bradley) [reply pkg/ottl/ottlfuncs/README.md]: This reads much better, thanks.

File: pkg/ottl/ottlfuncs/README.md (hunk 2 above)
Comment (evan-bradley) [reply pkg/ottl/ottlfuncs/README.md]: We did have one right below this paragraph, but each section was essentially reiterating the same information, so I consolidated and reworded them.

Comment (TylerHelmuth) [reply pkg/ottl/ottlfuncs/README.md]: I'm blind 👍

**Review by TylerHelmuth — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**